nmpolicy/internal/resolver/resolver_test.go

```diff
@@ -73,17 +77,38 @@ interfaces:
 `
 
 type test struct {
+	captureExpressions     string
 	captureASTPool         string
 	capturedStatesCache    string
 	expectedCapturedStates string
 	err                    string
 }
 
-func runTest(t *testing.T, testToRun test) {
+func withCaptureExpressions(t *testing.T, captureExpressions string) test {
+	testToRun := test{}
+	captureASTPool := types.CaptureASTPool{}
+	for captureEntryName, captureEntryExpression := range typestest.ToCaptureExpressions(t, captureExpressions) {
+		l := lexer.New()
+		tokens, err := l.Lex(captureEntryExpression)
+		assert.NoError(t, err)
+		p := parser.New()
+		astRoot, err := p.Parse(captureEntryExpression, tokens)
+		assert.NoError(t, err)
+		captureASTPool[captureEntryName] = astRoot
+	}
+	captureASTPoolMarshaled, err := yaml.Marshal(captureASTPool)
```

Comment:
Why marshalling the pool is required if we unmarshal it anyway in the `runTest`?

Reply:
So we don't have to change the rest of the code, since it's a test is not that problematic.

Reply:
What rest of the code?
Do you just need to change in `runTest`-
`captureASTPool := typestest.ToCaptureASTPool(t, testToRun.captureASTPool)` to
`captureASTPool := testToRun.captureASTPool`

Reply:
then we have to change all the test that are using "captureASTPool" as a string.